internal/controllers/topology/cluster/desired_state.go

```diff
@@ -776,19 +783,20 @@ func computeMachineDeploymentVersion(s *scope.Scope, machineDeploymentTopology c
 	// Return early if the currentVersion is already equal to the desiredVersion
 	// no further checks required.
 	if currentVersion == desiredVersion {
-		return currentVersion, nil
+		return currentVersion
 	}
 
 	// Return early if the upgrade for the MachineDeployment is deferred.
 	if isMachineDeploymentDeferred(s.Blueprint.Topology, machineDeploymentTopology) {
```

Comment:
I think the idea was that we also block new MD creations if the upgrade is deferred

I might be misremembering though. let's talk about it again